src/audio/AudioWorld.cpp

```diff
@@ -123,116 +100,42 @@ namespace World
 
             m_musicContext->provideLoader(loader);
 
-            for (const auto& segment : Utils::getFilesInDirectory(musicPath, "sgt")) {
+            for (const auto& segment : Utils::getFilesInDirectory(musicPath, "sgt"))
+            {
                 const auto lowercaseName = Utils::lowered(Utils::stripFilePath(segment));
                 const auto segm = m_musicContext->loadSegment(segment);
                 LogInfo() << "Loading " + segment;
                 m_Segments[lowercaseName] = m_musicContext->prepareSegment(*segm);
             }
             LogInfo() << "All segments loaded.";
 
-            alGenBuffers(RE_NUM_MUSIC_BUFFERS, m_musicBuffers);
-            alGenSources(1, &m_musicSource);
+            m_MusicSource = m_Engine.getAudioEngine().createSound(
+                [&](std::int16_t* buf, std::size_t len) -> int {
```

Comment:
nit: as types of `buf` and `len` are directly needed in lambda, you can use `auto`.